src/Ormolu/Printer/Meat/Type.hs

```diff
@@ -100,14 +101,28 @@ p_hsType = \case
     case p of
       Promoted -> txt "'"
       NotPromoted -> return ()
-    brackets . velt $ withSep comma (located' p_hsType) xs
+    brackets $ do
+        prefixForPromotedListTy xs
+        velt $ withSep comma (located' p_hsType) xs
   HsExplicitTupleTy NoExt xs -> do
     txt "'"
     parens . velt $ withSep comma (located' p_hsType) xs
   HsTyLit NoExt t -> atom t
   HsWildCardTy NoExt -> txt "_"
   XHsType (NHsCoreTy t) -> atom t
 
+prefixForPromotedListTy :: [LHsType GhcPs] -> R ()
+prefixForPromotedListTy (l@(L _ _):_) = located l prefix
+  where
+    prefix (HsTyVar _ Promoted _) = space
+    prefix _                      = return ()
+prefixForPromotedListTy _ = return ()
+prefixForPromotedTy = located' (\name ->
+ if last (showOutputable name) == '\'' then
```

Comment:
Looking at the `last` character is not going to be enough I don't think.
This is valid Haskell:

```haskell
{-# LANGUAGE DataKinds      #-}
{-# LANGUAGE KindSignatures #-}

data A = A'A

data B (a :: A) = B

b :: B ' A'A
b = B
```

Reply:
Checking if the second character is a `'` might be good enough! :)

```haskell
case showOutputable name of
  _:'\'':_ -> space
  _ -> return ()
```

Reply:
Thanks!
I see why checking for the last character won't work, but what about:
```haskell
data A = VeryLong' Name' 
```
perhaps we should check if the first non-alphanumeric character is a `'`?